Tools/C_util/Convergence/ComparePlotfiles.cpp

```diff
@@ -0,0 +1,253 @@
+
+#include <fstream>
+#include <iostream>
+
+#include <AMReX_ParmParse.H>
+#include <AMReX_PlotFileUtil.H>
+
+using namespace amrex;
+
+static
+void
+PrintUsage (const char* progName)
+{
+    std::cout << std::endl
+              << "This utility performs a diff operation between two"           << std::endl
+              << "plotfiles that have the same geometrical domain and nodality" << std::endl
+              << "(supports all nodality types; cell, face, edge, node)"        << std::endl
+              << "but possibly a factor of refinement between the cells,"       << std::endl
+              << "and outputs the L0, L1, and L2 norms"                         << std::endl
+              << "L1 = sum(|diff_ijk|)/npts_coarsedomain"                       << std::endl
+              << "L2 = sqrt[sum(diff_ijk^2)]/sqrt(npts_coarsedomain)"           << std::endl
+              <<  "(only single-level supported)"                               << std::endl << std::endl;
+    
+    std::cout << "Usage:" << '\n';
+    std::cout << progName << '\n';
+    std::cout << "    infile1 = inputFileName1" << '\n';
+    std::cout << "    reffile = refinedPlotFile" << '\n';
+    std::cout << "    diffile = differenceFileName" << '\n';
+    std::cout << "              (If not specified no file is written)" << '\n' << '\n';
+    
+    std::cout << "You can either point to the plotfile base directory itself, e.g."      << std::endl
+              << "  infile=plt00000"                                                     << std::endl
+              << "Or the raw data itself, e.g."                                          << std::endl
+              << "  infile=plt00000/Level_0/Cell"                                        << std::endl
+              << "the latter is useful for some applications that dump out raw"          << std::endl
+              << "nodal data within a plotfile directory."                               << std::endl
+              << "The program will first try appending 'Level_0/Cell'"                   << std::endl
+              << "onto the specified filenames."                                         << std::endl
+              << "If that _H file doesn't exist, it tries using the full specified name" << std::endl << std::endl;
+        
+    exit(1);
+}
+
+int
+main (int   argc,
+      char* argv[])
+{
+    amrex::Initialize(argc,argv);
+
+    if (argc == 1) {
```

Comment:
Maybe only the I/O process should call `PrintUsage`.

Reply:
done.  I changed std::cout to Print(); wrapping the function call in IOProcessor caused the remaining MPI ranks to continue with the program and spit out the wrong error messages about not specifying infile1